crates/workspace/src/workspace.rs

```diff
@@ -1563,14 +1563,17 @@ impl Workspace {
             multiple: true,
         });
 
+        let open_new_workspace_in_current_window =
+            WorkspaceSettings::get_global(cx).open_new_workspace_in_current_window;
+
         cx.spawn(|this, mut cx| async move {
             let Some(paths) = paths.await.log_err().flatten() else {
                 return;
             };
 
             if let Some(task) = this
                 .update(&mut cx, |this, cx| {
-                    this.open_workspace_for_paths(false, paths, cx)
+                    this.open_workspace_for_paths(open_new_workspace_in_current_window, paths, cx)
```

Comment:
I think you can just move the following code inside the update block, so you also only call `get_global` when you actually need it.

```rust
WorkspaceSettings::get_global(cx).open_new_workspace_in_current_window
```

Like so:

```rust
this.open_workspace_for_paths(
    WorkspaceSettings::get_global(cx).open_new_workspace_in_current_window, 
    paths, 
    cx
)
```

Reply:
Note that there are two more places where you should update the code to use the new setting:

1. https://github.com/zed-industries/zed/blob/f3e6a0beabb0ccb49d527ced03841a71098445e2/crates/recent_projects/src/recent_projects.rs#L307
2. https://github.com/zed-industries/zed/blob/f3e6a0beabb0ccb49d527ced03841a71098445e2/crates/recent_projects/src/recent_projects.rs#L315

Reply:
Fixed